kernel/src/main/java/org/kframework/unparser/ToLatex.java

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2019 K Team. All Rights Reserved.
+package org.kframework.unparser;
+
+import org.kframework.attributes.Att;
+import org.kframework.definition.Definition;
+import org.kframework.definition.Module;
+import org.kframework.definition.Sentence;
+import org.kframework.definition.Rule;
+import org.kframework.definition.ModuleComment;
+import org.kframework.definition.Production;
+import org.kframework.definition.SyntaxPriority;
+import org.kframework.definition.SyntaxSort;
+
+import org.kframework.kore.InjectedKLabel;
+import org.kframework.kore.K;
+import org.kframework.kore.KApply;
+import org.kframework.kore.KAs;
+import org.kframework.kore.KRewrite;
+import org.kframework.kore.KSequence;
+import org.kframework.kore.KToken;
+import org.kframework.kore.KVariable;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.io.IOException;
+import java.io.DataOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+
+import java.util.Optional;
+
+import scala.collection.JavaConverters;
+
+/**
+ * Writes a KAST term to the LaTeX format.
+ */
+public class ToLatex {
+
+    public static byte[] apply(K k) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), k);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K term to LaTeX", e, k);
+        }
+    }
+
+    public static String latexedKLabel(String orig) {
+        return "klabel" + orig.replaceAll("[^a-zA-Z]", "");
```

Comment:
this is not great because different klabels could be rewritten to the same latex macro. Take a look at how we deal with encoding of klabels in the ocaml and kore backends, then look at what characters are allowed in latex macro names and go from there.

Reply:
Taken care of.